### PR TITLE
RavenDB-8692 Fix after where clause completions

### PR DESCRIPTION
**Pull request by fitzchak**
Introduce query type which can be select or update.

### CODE REVIEW

File: src/Raven.Studio/typescript/common/queryCompleter.ts [hunk 1]
```diff
@@ -308,7 +309,9 @@ class queryCompleter {
         this.getIndexFields(queryIndexName.name, queryIndexName.type, prefix)
             .done((wordList) => {
                 if (functions) {
-                    wordList = wordList.concat(functions);
+                    functions.forEach(f => {
+                        wordList.push(f);
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/queryCompleter.ts]: You can use:
wordList.push(...functions)

File: src/Raven.Studio/typescript/common/queryCompleter.ts [hunk 2]
```diff
@@ -9,10 +9,11 @@ import collection = require("models/database/documents/collection");
 import document = require("models/database/documents/document");
 
 class queryCompleter {
+    private rules: AceAjax.RqlHighlightRules;
     private tokenIterator: new(session : AceAjax.IEditSession, initialRow: number, initialColumn: number) => AceAjax.TokenIterator = ace.require("ace/token_iterator").TokenIterator;
     private indexOrCollectionFieldsCache = new Map<string, autoCompleteWordList[]>();
     
-    constructor(private providers: queryCompleterProviders) {
+    constructor(private providers: queryCompleterProviders, private queryType: RqlQueryType) {
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/queryCompleter.ts]: I really like using strings enums here :)

File: src/Raven.Studio/typescript/common/queryCompleter.ts (hunk 1 above)
Comment (fitzchak) [reply src/Raven.Studio/typescript/common/queryCompleter.ts]: Thanks.

File: src/Raven.Studio/typescript/common/queryCompleter.ts (hunk 2 above)
Comment (fitzchak) [reply src/Raven.Studio/typescript/common/queryCompleter.ts]: Done

**Review by ml054 — APPROVED** [review]

**Pull request merged**